src/umockautoignoreargs.c

```diff
@@ -18,164 +20,215 @@ int umockautoignoreargs_is_call_argument_ignored(const char* call, size_t argume
 {
     int result;
 
-    if ((call == NULL) ||
-        (is_argument_ignored == NULL))
+    if (
+        (call == NULL) ||
+        (is_argument_ignored == NULL)
+        )
     {
         /* Codes_SRS_UMOCKAUTOIGNOREARGS_01_002: [ If call or is_argument_ignored is NULL, umockautoignoreargs_is_call_argument_ignored shall fail and return a non-zero value. ]*/
-        result = __LINE__;
+        result = MU_FAILURE;
     }
     else
     {
-        const char* cur_pos = strchr(call, '(');
+        // find where the args start
+        size_t call_string_len = strlen(call);
+        size_t i = call_string_len;
+        size_t paren_scope_count = 0;
+        bool is_error = false;
 
-        if (cur_pos == NULL)
+        /* Codes_SRS_UMOCKAUTOIGNOREARGS_01_010: [ umockautoignoreargs_is_call_argument_ignored shall look for the arguments as being the string between the last set of parenthesis in call. ]*/
+        do
         {
-            /* Codes_SRS_UMOCKAUTOIGNOREARGS_01_004: [ If umockautoignoreargs_is_call_argument_ignored fails parsing the call argument it shall fail and return a non-zero value. ]*/
-            result = __LINE__;
+            i--;
+            
+            if (call[i] == ')')
+            {
+                paren_scope_count++;
+            }
+            else if (call[i] == '(')
+            {
+                if (paren_scope_count == 0)
+                {
+                    // error
+                    is_error = true;
+                    break;
+                }
+                else
+                {
+                    paren_scope_count--;
+                    if (paren_scope_count == 0)
+                    {
+                        // found the start of the args
+                        break;
+                    }
+                }
+            }
+        } while (i > 0);
+
+        if (is_error)
```

Comment:
>is_error [](start = 12, length = 8)

Seems like `paren_scope_count` should be 0 here or there is an error, but the check is missing

Reply:
Added

---
In reply to: [375596015](https://github.com/Azure/umock-c/pull/140#discussion_r375596015) [](ancestors = 375596015)